src/main/java/pl/noip/lolstats/lol/stats/jwt/JwtParserImpl.java

```diff
@@ -28,8 +28,8 @@ public void setSecret(String secret) {
         this.secret = secret;
     }
 
-    @Override
-    public String getMail(String token) {
+
+    public Claims getData(String token) {
```

Comment:
This is not the best idea. Make this method private and invoke it in getName and getMail.